lib/active_model/serializer/attributes.rb

```diff
@@ -1,55 +1,23 @@
 module ActiveModel
   class Serializer
     module Attributes
-      # @api private
-      class Attribute
-        delegate :call, to: :reader
-
-        attr_reader :name, :reader
-
-        def initialize(name)
-          @name = name
-          @reader = :no_reader
-        end
-
-        def self.build(name, block)
-          if block
-            AttributeBlock.new(name, block)
-          else
-            AttributeReader.new(name)
-          end
-        end
-      end
-      # @api private
-      class AttributeReader < Attribute
-        def initialize(name)
-          super(name)
-          @reader = ->(instance) { instance.read_attribute_for_serialization(name) }
-        end
-      end
-      # @api private
-      class AttributeBlock < Attribute
-        def initialize(name, block)
-          super(name)
-          @reader = ->(instance) { instance.instance_eval(&block) }
-        end
-      end
-
       extend ActiveSupport::Concern
 
       included do
         with_options instance_writer: false, instance_reader: false do |serializer|
-          serializer.class_attribute :_attribute_mappings # @api private : maps attribute key names to names to names of implementing methods, @see #attribute
-          self._attribute_mappings ||= {}
+          serializer.class_attribute :_attributes_data # @api private
+          self._attributes_data ||= {}
```

Comment:
I still think it maps the serialization's attribute key to its value (or something that can get its value). But, I won't hold the PR for that. Too bad no one else is commenting :disappointed: 

Code looks good.  Just:

- Mind adding a changelog
- [you're right about the location of the autoload](https://github.com/rails-api/active_model_serializers/pull/1370/files#r48460097). I somehow saw the ActiveSupport::Concern and read it as ActiveSupport::Autoload.  In any case, we should use `extend ActiveSupport::Autoload`, is what I meant.  It does have a slightly terser (convention-driven) syntax available, but also works with eager_loading, etc.  And since it *doesn't* work with 'require', I'd rather just be consistent. http://api.rubyonrails.org/classes/ActiveSupport/Autoload.html